src/matches.rs

```diff
@@ -27,7 +27,8 @@ macro_rules! matches {
 }
 
 /// Work around "error: unexpected token: `an interpolated tt`", whatever that
-/// means.
+/// means. (Probably rust-lang/rust#22819.)
+#[doc(hidden)]
```

Comment:
There's an unimplemented part of [macro reform](https://github.com/rust-lang/rfcs/blob/master/text/0453-macro-reform.md) which allows an exported macro to list some private "helper macros".

I believe `_tt_as_expr_hack` itself won't be necessary once rust-lang/rust#22819 is fixed.